### PR TITLE
A lot of changes to BottomSheet to make it more flexible

### PR DESCRIPTION
**Pull request by haavamoa**
### Description of Change

- [BottomSheet] Will now respond to property changes when its open.
- [BottomSheet] Added new `Position` property which will determine what position the bottom sheet will present in when opened.
- [BottomSheet] People can drag the bottom sheet to large when the position is medium and vica versa.
- [BottomSheet] Consumer can open multiple bottom sheets and close them by using `BottomSheetService.CloseAll()`
- [BottomSheet][BreakingChange] Replaced `OpenBottomSheet()` with `Open()`
- [BottomSheet][BreakingChange] ShouldFitContent is replaced with `Position` = `Fit`

### Todos
- [X] I have tested on an Android device.
- [X] I have tested on an iOS device.
- [ ] I have supported accessibility

### CODE REVIEW

File: src/library/DIPS.Mobile.UI/Components/BottomSheets/iOS/BottomSheetContentPage.cs [hunk 1]
```diff
@@ -10,12 +10,12 @@ namespace DIPS.Mobile.UI.Components.BottomSheets.iOS;
...
 {
-    private UIViewController? m_viewController;
-#nullable disable
-    private UIViewController m_navigationViewController;
-#nullable enable
+
     private readonly BottomSheet m_bottomSheet;
-    private UISheetPresentationController? m_sheetPresentationController;
+
+    // public UIViewController UINavigationController { get; set; }
```
Comment (Vetle444) [inline src/library/DIPS.Mobile.UI/Components/BottomSheets/iOS/BottomSheetContentPage.cs:16]: Fjern?

File: src/library/DIPS.Mobile.UI/Components/BottomSheets/iOS/BottomSheetContentPage.cs [hunk 2]
```diff
@@ -35,6 +35,44 @@ public BottomSheetContentPage(BottomSheet bottomSheet)
...
+        }
+        else
+        {
+            navigationController = page.ToUIViewController(mauiContext);
+        }
+        
+        var uiSheetPresentationController = navigationController.SheetPresentationController;
+        
+        //TODO: MOVE TO HANDLER:
+        //uiSheetPresentationController!.Delegate = new BottomSheetControllerDelegate(this);
```
Comment (Vetle444) [inline src/library/DIPS.Mobile.UI/Components/BottomSheets/iOS/BottomSheetContentPage.cs:70]: Gjør du detta seinar?

File: src/library/DIPS.Mobile.UI/Components/BottomSheets/iOS/BottomSheetContentPage.cs [hunk 3]
```diff
@@ -52,53 +90,50 @@ private void IncludeSearchBar()
...
-            m_navigationViewController.ModalInPresentation = true;
-        }
-        
-        if(m_bottomSheet.ShouldHaveNavigationBar)        
-            ConfigureToolbar();
-        
-        m_sheetPresentationController = m_navigationViewController.SheetPresentationController;
-        m_sheetPresentationController!.Delegate = new BottomSheetControllerDelegate(this);
-
+        // if (!OperatingSystem.IsIOSVersionAtLeast(15))
```
Comment (Vetle444) [inline src/library/DIPS.Mobile.UI/Components/BottomSheets/iOS/BottomSheetContentPage.cs:93]: .

File: src/library/DIPS.Mobile.UI/Components/BottomSheets/iOS/BottomSheetContentPage.cs [hunk 4]
```diff
@@ -153,73 +188,26 @@ private void SetupViewController()
...
-    private void ConfigureToolbar()
+    private static void ConfigureToolbar(UIViewController uiViewController)
     {
-        Title = m_bottomSheet.Title;
-        foreach (var item in m_bottomSheet.ToolbarItems)
-        {
-            item.BindingContext = m_bottomSheet.BindingContext;
-            ToolbarItems.Add(item);
-        }
+        //TODO: Move to handler
```
Comment (Vetle444) [inline src/library/DIPS.Mobile.UI/Components/BottomSheets/iOS/BottomSheetContentPage.cs:193]: Samme som over

File: src/library/DIPS.Mobile.UI/Components/BottomSheets/BottomSheet.cs [hunk 5]
```diff
@@ -15,12 +15,52 @@ public BottomSheet()
...
+        {
+            return BottomSheetService.Open(this);
         }
-        
-        internal SearchBar? SearchBar { get; private set; }
-        
-        internal bool ShouldHaveNavigationBar => !string.IsNullOrEmpty(Title) || ToolbarItems is { Count: > 0 };
-        
+
+        private void OnUnLoaded(object? sender, EventArgs e)
```
Comment (Vetle444) [inline src/library/DIPS.Mobile.UI/Components/BottomSheets/BottomSheet.cs:54]: Vi fant jo ut at UnLoaded blir kjørt vess man naviger bort, lura på om vess AuthenticationPage kommer over denna, så kjøres UnLoaded?

Comment (Vetle444) [reply src/library/DIPS.Mobile.UI/Components/BottomSheets/BottomSheet.cs:54]: Eller vess man seinare adda inn navigering i BottomSheet

Comment (Vetle444) [reply src/library/DIPS.Mobile.UI/Components/BottomSheets/BottomSheet.cs:54]: Trur rektige måten e å override OnHandlerChanging

Comment (haavamoa) [reply src/library/DIPS.Mobile.UI/Components/BottomSheets/BottomSheet.cs:54]: Dessverre ingen event på HandlerChanging når viewet forkastes, skal sjekke ute andre ting.

Comment (haavamoa) [reply src/library/DIPS.Mobile.UI/Components/BottomSheets/BottomSheet.cs]: Løste dette ved å disconnect handler selv, tror man må påse dette selv og at det ikke kommer ut av boksen når mand dismisser parent.

**Review by Vetle444 — APPROVED** [review]

**Pull request merged**